src/lava/magma/compiler/builder.py

```diff
@@ -359,11 +359,12 @@ def build(self):
         if self.proc_params:
             pm = self.proc_model(self.proc_params)
         else:
-            # Empty argument because some existing process models where __init__
-            # is overloaded without proc_model argument, the execution hangs.
-            # However, when the base class constructor is called, proc_model
-            # is initialized to a default value of {} which is the desired
+            # Calling base class constructor directly where proc_params is
+            # initialized to a default value of {} which is the desired
             # behavior.
+            # This enables backward compatibility with some existing processes
```

Comment:
I understand that this would have triggered more work right now. But by enabling the proc_params feature you uncovered a deficiency in the older __init__. 
If you now introduce such a hack, then we should at least file an issue to fix the underlying __init__ issue later.
Ideally, if not too much work, we should fix it right away.

Reply:
fixed